app/src/main/java/com/ably/tracking/demo/subscriber/domain/OrderState.kt

```diff
@@ -0,0 +1,12 @@
+package com.ably.tracking.demo.subscriber.domain
+
+data class OrderState(
+    val state: State,
+    val errorMessage: String?
+) {
+    enum class State {
+        Online, Publishing, Offline, Failed
```

Comment:
Nothing blocking but just a question: Why do those states directly correspond to to the same state as trackable state?

Reply:
I have created our own model of states based on the trackable state to reduce coupling to AAT SDK cod, limit it to "lower" layers and make it easier to use in "upper" layers.
In this case, it simplified displaying error state a bit by moving traversing SDK models to the `OrderManager`.

![Screenshot 2022-08-25 at 15 07 50](https://user-images.githubusercontent.com/10450116/186673136-6b0a1a0a-424d-4827-aa38-c461407aa03d.png)